includes/Feed/FeedProductFormatter.php

```diff
@@ -0,0 +1,236 @@
+		);
+		$feed_row['availability']              = $this->get_value_from_product_data( $product_data, 'availability' );
+		$feed_row['item_group_id']             = $item_group_id;
+		$feed_row['checkout_url']              = $this->get_value_from_product_data( $product_data, 'checkout_url' );
+		$feed_row['additional_image_link']     = $this->format_additional_image_url( $this->get_value_from_product_data( $product_data, 'additional_image_urls' ) );
+		$feed_row['sale_price_effective_date'] = $this->get_value_from_product_data( $product_data, 'sale_price_start_date' ) . '/' . $this->get_value_from_product_data( $product_data, 'sale_price_end_date' );
+		$feed_row['sale_price']                = $this->format_price_for_feed(
+			$this->get_value_from_product_data( $product_data, 'sale_price', 0 ),
+			$this->get_value_from_product_data( $product_data, 'currency' )
+		);
+		$feed_row['condition']                 = 'new';
+		$feed_row['visibility']                = $this->get_value_from_product_data( $product_data, 'visibility' );
+		$feed_row['gender']                    = $this->get_value_from_product_data( $product_data, 'gender' );
+		$feed_row['color']                     = $this->get_value_from_product_data( $product_data, 'color' );
+		$feed_row['size']                      = $this->get_value_from_product_data( $product_data, 'size' );
+		$feed_row['pattern']                   = $this->get_value_from_product_data( $product_data, 'pattern' );
+		$feed_row['google_product_category']   = $this->get_value_from_product_data( $product_data, 'google_product_category' );
+		$feed_row['default_product']           = $this->get_value_from_product_data( $product_data, 'default_product' );
+		$feed_row['variant']                   = $this->get_value_from_product_data( $product_data, 'variant' );
+
+		return $feed_row;
+	}
+
+	/**
+	 * Prepare variation information
+	 *
+	 * @since 2.6.0
+	 * @param array                $product_data Product information for the feed.
+	 * @param \WC_Facebook_Product $woo_product WooCommerce product object normalized by Facebook.
+	 * @param array                $attribute_variants Array of variants attributes.
+	 * @return array[string] product feed data
+	 */
+	private function prepare_variation_for_feed( $product_data, $woo_product, &$attribute_variants ) {
+		$parent_id = $woo_product->get_parent_id();
+
+		if ( ! isset( $attribute_variants[ $parent_id ] ) ) {
+
+			$parent_product          = new \WC_Facebook_Product( $parent_id );
+			$gallery_urls            = array_filter( $parent_product->get_gallery_urls() );
+			$variation_id            = $parent_product->find_matching_product_variation();
+			$variants_for_group      = $parent_product->prepare_variants_for_group( true );
+			$parent_attribute_values = array(
+				'gallery_urls'       => $gallery_urls,
+				'default_variant_id' => $variation_id,
+				'item_group_id'      => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $parent_product ),
```

Comment:
My IDE is flagging an issue here because `get_fb_retailer_id` expects an instance of `WC_Product` but `WC_Facebook_Product` is being passed.